src/ServiceWire/NamedPipes/NpListener.cs

```diff
@@ -14,17 +12,19 @@ public class NpListener
         private int _maxConnections = 254;
         private ILog _log = new NullLogger();
         private IStats _stats = new NullStats();
+        private INamedPipeServerStreamFactory _streamFactory = new DefaultNamedPipeServerStreamFactory();
 
         public string PipeName { get; set; }
         public event EventHandler<PipeClientConnectionEventArgs> RequestReieved;
 
-        public NpListener(string pipeName, int maxConnections = 254, ILog log = null, IStats stats = null)
+        public NpListener(string pipeName, int maxConnections = 254, ILog log = null, IStats stats = null, INamedPipeServerStreamFactory streamFactory = null)
         {
             _log = log ?? _log;
             _stats = stats ?? _stats;
             if (maxConnections > 254) maxConnections = 254;
             _maxConnections = maxConnections;
             this.PipeName = pipeName;
+            _streamFactory= streamFactory ?? _streamFactory;
```

Comment:
Rather than create a new default in the initializer, let's create the default here only if necessary.

Reply:
Sure thing, I'll change that.
I actually did it because the `_log`, `_stats` and `_maxConnections` variables all do it. Would you like me to change those too, or leave them as they are?